src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/test_role.py

```diff
@@ -37,29 +38,55 @@ def body(self):
             JMESPathCheck('properties.type', 'BuiltInRole')
         ])
 
-class RoleAssignmentScenarioTest(VCRTestBase):
-
-    def test_role_assignment_scenario(self):
-        self.execute()
+class RoleAssignmentScenarioTest(ResourceGroupVCRTestBase):
 
     def __init__(self, test_method):
         super(RoleAssignmentScenarioTest, self).__init__(__file__, test_method)
+        self.resource_group = 'cli-role-assignment-test'
+        self.user = 'testuser1@azuresdkteam.onmicrosoft.com'
 
-    def body(self):
-        s = self
-        principal_id = '7a938a30-4226-420e-996f-4d48bca6d537'
-        scope = '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590'
-        rg = 'travistestresourcegroup'
-        resource = 'testsql23456/'
-        parent_path = 'servers/testserver23456'
+    def set_up(self):
+        super().set_up()
+        self.cmd('ad user create --display-name tester123 --password Test123456789 --user-principal-name {}'.format(self.user), None)
+        time.sleep(15) #By-design, it takes some time for RBAC system propagated with graph object change
```

Comment:
Yep, it is mocked out. The code is [here](https://github.com/Azure/azure-cli/blob/master/src/azure/cli/utils/vcr_test_base.py#L246)